gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireStore.java

```diff
@@ -2308,6 +2299,15 @@ public void setDBName(String dbname) {
         this.database.setdisableStatementOptimizationToGenericPlan();
       }
     }
+    // install the GemFireXD specific thread dump signal (URG) handler
+    if (!isSnappyStore()) {
+      try {
+        SigThreadDumpHandler.install();
+      } catch (Throwable t) {
+        SanityManager.DEBUG_PRINT("fine:TRACE",
+            "Failed to install thread dump signal handler: " + t.getCause());
+      }
+    }
   }
 
```

Comment:
Ideally we would like the handler to work with Spark too. Can you file a JIRA to explore if some other signal works with Spark too: SIGCHLD, SIGWINCH, SIGPWR, SIGIO. Last two may not be available on most UNIX systems, so we also need to select one which is available on the UNIX platforms we will be interested in going forward: Linux, Mac, Solaris